### ISSUE TITLE
Error: Cannot find module 'async'

### ISSUE THREAD

**Issue by Macxim** [lipsumar/dsf-example-project#1]
When running `npm run dsf` for the first time, I get this:

``` zsh
~/code/dsf-example-project master  
❯ npm run dsf

> dsf-example-project@0.0.0 dsf /Users/macxim/code/dsf-example-project
> node node_modules/design-system-framework/cli.js

   ⌜                         ⌝
     Design System Framework  
   ⌞                         ⌟

module.js:338
    throw err;
    ^
Error: Cannot find module 'async'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/Users/macxim/code/dsf-example-project/node_modules/design-system-framework/models/ComponentCandidate.js:4:10)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)

npm ERR! Darwin 15.4.0
npm ERR! argv "/Users/macxim/.nvm/versions/node/v0.12.7/bin/node" "/Users/macxim/.nvm/versions/node/v0.12.7/bin/npm" "run" "dsf"
npm ERR! node v0.12.7
npm ERR! npm  v2.11.3
npm ERR! code ELIFECYCLE
npm ERR! dsf-example-project@0.0.0 dsf: `node node_modules/design-system-framework/cli.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the dsf-example-project@0.0.0 dsf script 'node node_modules/design-system-framework/cli.js'.
npm ERR! This is most likely a problem with the dsf-example-project package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node node_modules/design-system-framework/cli.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls dsf-example-project
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/macxim/code/dsf-example-project/npm-debug.log

```

FYI:

``` zsh
❯❯ npm -v
2.14.2                                                                                                                                                          
❯❯ node -v
v4.0.0
```

Then, I just `npm install async`, and I got:

``` console
path.js:8
    throw new TypeError('Path must be a string. Received ' +
    ^

TypeError: Path must be a string. Received undefined
    at assertPath (path.js:8:11)
    at Object.posix.join (path.js:477:5)
    at ComponentBase.getGlobPath (/Users/macxim/code/dsf-example-project/node_modules/design-system-framework/models/ComponentBase.js:12:14)
    at ComponentBase.ComponentCandidate.hasResource (/Users/macxim/code/dsf-example-project/node_modules/design-system-framework/models/ComponentCandidate.js:24:12)
    at /Users/macxim/code/dsf-example-project/node_modules/async/lib/async.js:356:13
    at async.forEachOf.async.eachOf (/Users/macxim/code/dsf-example-project/node_modules/async/lib/async.js:233:13)
    at _asyncMap (/Users/macxim/code/dsf-example-project/node_modules/async/lib/async.js:355:9)
    at Object.map (/Users/macxim/code/dsf-example-project/node_modules/async/lib/async.js:337:20)
    at ComponentBase.ComponentCandidate.resolve (/Users/macxim/code/dsf-example-project/node_modules/design-system-framework/models/ComponentCandidate.js:14:8)
    at addComponentPathCandidate (/Users/macxim/code/dsf-example-project/node_modules/design-system-framework/lib/dsf.js:72:12)
```

**Comment by lipsumar** [comment]
Indeed, dependency `async` is missing. You fixed it correctly. Fix is on the way https://github.com/lipsumar/design-system-framework/issues/2

Did you modify the example project ?

**Comment by Macxim** [comment]
No, I didn't modify anything. I just followed the steps in the README.

**Comment by lipsumar** [comment]
I'll try that again with node v4. I developed and tested with 0.12.4 only so far.

**Comment by Macxim** [comment]
I got the same result with `0.12.4`.

**Comment by lipsumar** [comment]
yep, i get that too. I believe this is because the user config is not correctly merged.
It was fixed in design-system-framework 0.1.1
I forgot to update the example project, so you are running 0.1.0
I just pushed the fix to the example project.

Please run:
`$ git pull`
`$ npm install`
`$ npm run dsf`

**Comment by Macxim** [comment]
Thanks :)
Works perfectly now.
